optional/nothing.py

```diff
@@ -1,43 +1,59 @@
 from .abstract_optional import AbstractOptional
 from .exceptions import OptionalAccessOfEmptyException
+from typing import TypeVar, Generic, Callable, NoReturn
 
 
-class Nothing(AbstractOptional):
+T = TypeVar('T')
```

Comment:
I couldn't get PyCharm to tell me what my return type would be but maybe I am using it wrong.

Reply:
![Screen Shot 2020-06-23 at 11 30 40 AM](https://user-images.githubusercontent.com/554207/85429912-076a4f80-b545-11ea-9c1a-a2319b2e8e20.png)

Reply:
As you can see above, it was able to tell that: 

1. `a` is either a `Something` or a `Nothing` but isnt capable of telling which.
2. `x` is a bool, it also type hinted to me that I should pass an AbstractOptional to the `__eq__` method

However it was unable to tell what type `b` was.